src/app/recipes/components/recipe-details/recipe-details.component.ts

```diff
@@ -54,9 +64,113 @@ export class RecipeDetailsComponent implements OnInit, OnDestroy {
         } else {
             this.user.favoriteRecipes = [...this.user.favoriteRecipes, this.recipe.id];
         }
+
+        this.recipeBookmarked = this.isRecipeBookmarked();
+    }
+
+    isRecipeBookmarked() {
+       return this.user.favoriteRecipes.includes(this.recipe.id);
+    }
+
+    scroll(el: HTMLElement) {
+        el.scrollIntoView({behavior: 'smooth'});
+    }
+
+    onCheckIngredient(category: string, ingredient: string) {
+        let ingredients: CheckedRecipeIngredients[] = [];
+        if (this.isLocalStorageAvailable) {
```

Comment:
Why do you need this check? Is it because of server side rendering? You should check e.g. onInit if it's client side rendering (isPlatformBrowser) and only then restore selection. And get rid of this flag